p2p/src/net/default_backend/backend.rs

```diff
@@ -523,63 +570,64 @@ where
         Ok(())
     }
 
-    async fn handle_command(&mut self, command: Command<T::Address>) -> crate::Result<()> {
-        // All handlings are separated to two parts:
+    fn handle_command(&mut self, command: Command<T::Address>) {
+        // All handlings can be separated to two parts:
         // - Async (can't take mutable reference to self because they are run concurrently).
         // - Sync (take mutable reference to self because they are run sequentially).
         // Because the second part depends on result of the first part boxed closures are used.
 
-        let backend_task: BackendTask<T> = match command {
+        match command {
             Command::Connect { address } => {
                 let connection_fut = timeout(
                     *self.p2p_config.outbound_connection_timeout,
                     self.transport.connect(address.clone()),
                 );
 
-                async move {
+                let backend_task: BackendTask<T> = async move {
                     let connection_res = connection_fut.await.unwrap_or(Err(P2pError::DialError(
                         DialError::ConnectionRefusedOrTimedOut,
                     )));
 
                     boxed_cb(move |this| this.handle_connect_res(address, connection_res))
                 }
-                .boxed()
+                .boxed();
+
+                self.command_queue.push(backend_task);
             }
-            Command::Disconnect { peer_id } => async move {
-                boxed_cb(move |this: &mut Self| {
```

Comment:
Why are we removing these?

Reply:
This code was added when the channels were bounded and sending required `await`. But after later refactoring, when we switched to unbounded channels, this code became just overhead.
For example, for each received message, we first construct a future, push it into the `FuturesUnordered` list, and this future is immediately polled on a next event loop iteration.